### PR TITLE
InfectSweep - extract helper functions to add infections

### LINKED ISSUE #438: Sweep.cpp contains large functions and duplicated code
There is room for consolidation - some of the logic can be factored out into common functions.
Take each kind of sweep and apply refactorings to eliminate the duplicated code.
Fix #435

### PR DESCRIPTION
**Pull request by mstevens-uk**
Fixes #438 

extract helper functions for logic in InfectSweep

void AddToInfectionQueue(const int tn, const int infectee_cell_number, const int infector_index, const int infectee_index, const short int infect_type);

bool AddInfections(const int tn, const int infectee_cell_index, const int infector_index, const int infectee_index, const short int infect_type);

### CODE REVIEW

**Comment by mstevens-uk** [comment]
@dlaydon @weshinsley Just realised this change to InfectSweep affects the DctQueue - and have been instructed to leave Dct alone, I'll revert the Dct bit and resubmit.

**Comment by mstevens-uk** [comment]
@weshinsley @matt-gretton-dann @dlaydon This PR is now ready for review/merge. It does not touch Digital Tracing, and removes  duplicate code concerning the infection queue. Parameters are renamed sensibly, asserts are added, and comments are included.

File: src/Sweep.cpp [hunk 1]
```diff
@@ -533,7 +525,7 @@ void InfectSweep(double t, int run) //added run number as argument in order to r
 										// allow care home residents to mix more intensely in "groups" (i.e. individual homes) than staff do - to allow for PPE/environmental contamination.
 										if ((k==P.CareHomePlaceType)&&((!Hosts[ci].care_home_resident)||(!Hosts[i3].care_home_resident))) s *= P.CareHomeWorkerGroupScaling;
 										//these are all place group contacts to be tracked for digital contact tracing - add to StateT queue for contact tracing
-										//if infectee is also a user, add them as a contact
+										//if infectee is also a user, add them as a contact
```
[the remark this answers is not in the document]
Comment (mstevens-uk) [reply src/Sweep.cpp:528]: done

File: src/Sweep.cpp [hunk 2]
```diff
@@ -665,7 +648,7 @@ void InfectSweep(double t, int run) //added run number as argument in order to r
 												}
 											}
 										}
-
+
```
[the remark this answers is not in the document]
Comment (mstevens-uk) [reply src/Sweep.cpp:651]: done

File: src/Sweep.cpp [hunk 3]
```diff
@@ -2010,3 +1981,64 @@ int TreatSweep(double t)
...
+ * @param infectee_cell_index -  cell number of potential infectee
+ * @param infector_index
+ * @param infectee_index
+ * @param infect_type
+ * @return if the queue is added to, return true
+ */
+bool AddInfections(const int tn, const int infectee_cell_index, const int infector_index, const int infectee_index,
+	const short int infect_type)
+{
+	bool validEntryAddedToQueue = false;
```
[the remark this answers is not in the document]
Comment (mstevens-uk) [reply src/Sweep.cpp:2015]: done

**Comment by mstevens-uk** [comment]
Hi @weshinsley. Your feedback has been addressed. Re: performance, I haven't noticed any impact on running times based on the standard regression tests. All params to the new functions are plain ints. I would be interested in the results on a large run.

**Comment by weshinsley** [comment]
No noticeable impact - forgive the paranoia...

**Pull request merged**